### PR TITLE
Improved Default Priors for `BetaGeoModel`

### LINKED ISSUE #1123: Update `default_model_config` for `BetaGeoModel`
`BetaGeoModel` utilizes HalfFlat default priors which are extraordinarily slow for model fitting. Gamma priors would work well for `r_prior` and `alpha_prior`, but `a_prior` and `b_prior` would benefit from a pooling approach:

https://mc-stan.org/users/documentation/case-studies/pool-binary-trials.html

This has already been implemented for `BetaGeoBetaBinomModel`, so most of the required work for this could be copy/pasted over:

https://github.com/pymc-labs/pymc-marketing/blob/main/pymc_marketing/clv/models/beta_geo_beta_binom.py#L175

### PR DESCRIPTION
**Pull request by ColtAllen**
## Description

This PR modifies the default model config of `BetaGeoModel` with a prior specification for the dropout process similar to that in `BetaGeoBetaBinomModel`.  This improved parameter estimates and reduced fit times by 40-50% in testing.

## Related Issue

- [x] Closes #1123
- [x] Related to https://github.com/pymc-labs/pymc-marketing/issues/1077

## Checklist

- [x] Checked that [the pre-commit linting/style checks pass](https://docs.pymc.io/en/latest/contributing/python_style.html)
- [ ] Included tests that prove the fix is effective or that the new feature works
- [x] Added necessary documentation (docstrings and/or example notebooks)
- [ ] If you are a pro: each commit corresponds to a [relevant logical change](https://wiki.openstack.org/wiki/GitCommitMessages#Structural_split_of_changes)

## Modules affected

- [ ] MMM
- [x] CLV

## Type of change

- [x] New feature / enhancement
- [ ] Bug fix
- [x] Documentation
- [ ] Maintenance
- [ ] Other (please specify):

----
📚 Documentation preview 📚: https://pymc-marketing--1264.org.readthedocs.build/en/1264/

### CODE REVIEW

**Comment by wd60622** [comment]
can you add `graphviz` method instead of importing to close out #1138

File: docs/source/notebooks/clv/dev/MMM-Sandbox.ipynb
Comment (wd60622) [inline docs/source/notebooks/clv/dev/MMM-Sandbox.ipynb:1]: what's with these two mmm changes?

Comment (ColtAllen) [reply docs/source/notebooks/clv/dev/MMM-Sandbox.ipynb:1]: I created a sandbox branch for playing around with MMM/CLV ideas, and forgot which branch I was in 🫢

**Comment by ColtAllen** [comment]
> can you add `graphviz` method instead of importing to close out #1138

I'll create another PR for this prior to merging this one. If it can be added to `ModelBuilder` I'd prefer to remove the existing MMM method.

**Review by juanitorduz — APPROVED** [review]

**Comment by juanitorduz** [comment]
After addressing @wd60622 's comments we can merge this one :)

**Comment by ColtAllen** [comment]
https://github.com/pymc-labs/pymc-marketing/pull/1284

**Comment by ColtAllen** [comment]
> > can you add `graphviz` method instead of importing to close out #1138
> 
> I'll create another PR for this prior to merging this one. If it can be added to `ModelBuilder` I'd prefer to remove the existing MMM method.

https://github.com/pymc-labs/pymc-marketing/pull/1284

**Pull request merged**